grpc/src/main/java/com/linecorp/armeria/internal/client/grpc/GrpcClientFactory.java

```diff
@@ -136,18 +136,18 @@ public Object newClient(ClientBuilderParams params) {
             jsonMarshaller = null;
         }
 
-        Channel channel = new ArmeriaChannel(
-                newParams,
-                httpClient,
-                meterRegistry(),
-                scheme.sessionProtocol(),
-                serializationFormat,
-                jsonMarshaller,
-                simpleMethodNames);
+        final ArmeriaChannel armeriaChannel =
+                new ArmeriaChannel(newParams, httpClient, meterRegistry(), scheme.sessionProtocol(),
+                                   serializationFormat, jsonMarshaller, simpleMethodNames);
         final Iterable<? extends ClientInterceptor> interceptors = options.get(GrpcClientOptions.INTERCEPTORS);
+        final Channel channel;
         if (!Iterables.isEmpty(interceptors)) {
-            channel = ClientInterceptors.intercept(channel, Iterables.toArray(interceptors,
-                                                                              ClientInterceptor.class));
+            final Channel intercepted =
+                    ClientInterceptors.intercept(armeriaChannel,
+                                                 Iterables.toArray(interceptors, ClientInterceptor.class));
+            channel = new UnwrappableChannel(intercepted, armeriaChannel);
```

Comment:
Note: I think we can't rely on this behavior since users can also add interceptors after building the client.
I think this addition doesn't hurt though  👍

Reply:
> I think we can't rely on this behavior since users can also add interceptors after building the client.

`Unwrappable` is used to create a derived client.
https://github.com/line/armeria/blob/5b384fbe27e7e6f9225d6db91cbb684d09dfbb5e/core/src/main/java/com/linecorp/armeria/client/Clients.java#L258-L264
Originally, a client is intercepted after creation, it cannot be unwrapped. That is out of Armeria's scope. But I'd like to make the best effort we can. 😄